nexus/test-utils/src/lib.rs

```diff
@@ -1775,6 +1758,10 @@ async fn setup_with_config_impl<N: NexusServer>(
                             .boxed()
                     }),
                 ),
+                (
```

Comment:
I'm curious why this had to move to after starting Nexus.  This configuration mimics what RSS would normally do, right?  And that happens before Nexus starts.

Reply:
I had the blueprint-builder recording the disks & datasets it creates, with the sled-agent picking them up later. But since you asked (thanks for asking!), I hacked on it some more and got it working the other way: in 4c1f2ab01, `sled_agent_config` now calls a new `make_sled_configs` method that computes what will be the blueprint's sled configs, and uses the disks & datasets from that. This lets sled-agent start before Nexus again, and the configs to still agree.